diff -Nurd luasql/src/ls_mysql.c luasql/src/ls_mysql.c
--- luasql/src/ls_mysql.c	2016-04-14 06:25:33.675191715 +0300
+++ luasql/src/ls_mysql.c	2016-04-14 06:26:22.555141176 +0300
@@ -183,7 +183,7 @@
 /*
 ** Closes the cursos and nullify all structure fields.
 */
-static int cur_nullify (lua_State *L, cur_data *cur) {
+static void cur_nullify (lua_State *L, cur_data *cur) {
 	/* Nullify structure fields. */
 	cur->closed = 1;
 	mysql_free_result(cur->my_res);
