diff -Nurd tolua-5.2/src/bin/tolua.c tolua-5.2/src/bin/tolua.c
--- tolua-5.2/src/bin/tolua.c	2022-09-26 18:11:41.769267668 +0300
+++ tolua-5.2/src/bin/tolua.c	2020-02-22 14:33:47.775736302 +0200
@@ -22,6 +22,9 @@
 #include <stdlib.h>
 #include <string.h>
 
+#ifndef LUA_SOURCE
+int tolua_tolua_open(lua_State *L);
+#endif
 
 static void help (void)
 {
@@ -106,7 +109,6 @@
 
 #ifndef LUA_SOURCE
  {
-  int tolua_tolua_open (lua_State* L);
   tolua_tolua_open(L);
  }
 #else
