diff -Nurd lua-5.5/src/lcode.c lua-5.5/src/lcode.c
--- lua-5.5/src/lcode.c	2025-11-27 17:03:35.688793387 +0200
+++ lua-5.5/src/lcode.c	2025-11-27 17:06:30.816221321 +0200
@@ -14,6 +14,7 @@
 #include <limits.h>
 #include <math.h>
 #include <stdlib.h>
+#include <stdarg.h>
 
 #include "lua.h"
 
diff -Nurd lua-5.5/src/ldo.c lua-5.5/src/ldo.c
--- lua-5.5/src/ldo.c	2025-11-27 17:05:27.967558886 +0200
+++ lua-5.5/src/ldo.c	2025-11-21 06:41:40.432836170 +0200
@@ -92,7 +92,7 @@
 }
 
 
-#elif defined(LUA_USE_POSIX)				/* }{ */
+#elif defined(LUA_USE_ULONGJMP)				/* }{ */
 
 /* in POSIX, use _longjmp/_setjmp (more efficient) */
 #define LUAI_THROW(L,c)		_longjmp((c)->b, 1)
@@ -141,9 +141,9 @@
         lua_unlock(L);
         g->panic(L);  /* call panic function (last chance to jump out) */
       }
-      abort();
     }
   }
+  abort();
 }
 
 
diff -Nurd lua-5.5/src/liolib.c lua-5.5/src/liolib.c
--- lua-5.5/src/liolib.c	2025-11-27 17:05:27.967660107 +0200
+++ lua-5.5/src/liolib.c	2025-11-21 06:41:40.432836170 +0200
@@ -53,7 +53,7 @@
 
 #if !defined(l_popen)		/* { */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_POPEN)      /* { */
 
 #define l_popen(L,c,m)		(fflush(NULL), popen(c,m))
 #define l_pclose(L,file)	(pclose(file))
@@ -114,7 +114,7 @@
 
 #if !defined(l_fseek)		/* { */
 
-#if defined(LUA_USE_POSIX) || defined(LUA_USE_OFF_T)	/* { */
+#if defined(LUA_USE_FSEEKO) || defined(LUA_USE_OFF_T)	/* { */
 
 #include <sys/types.h>
 
diff -Nurd lua-5.5/src/loslib.c lua-5.5/src/loslib.c
--- lua-5.5/src/loslib.c	2025-11-27 17:05:27.967720867 +0200
+++ lua-5.5/src/loslib.c	2025-11-21 06:41:40.432836170 +0200
@@ -75,7 +75,7 @@
 ** where it uses gmtime_r/localtime_r
 */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_GMTIME_R)   /* { */
 
 #define l_gmtime(t,r)		gmtime_r(t,r)
 #define l_localtime(t,r)	localtime_r(t,r)
@@ -102,7 +102,7 @@
 */
 #if !defined(lua_tmpnam)	/* { */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_MKSTEMP)    /* { */
 
 #include <unistd.h>
 
diff -Nurd lua-5.5/src/luaconf.h lua-5.5/src/luaconf.h
--- lua-5.5/src/luaconf.h	2025-11-27 17:05:27.967803967 +0200
+++ lua-5.5/src/luaconf.h	2025-11-21 06:41:40.436836194 +0200
@@ -11,6 +11,7 @@
 #include <limits.h>
 #include <stddef.h>
 
+#include "localluaconf.h"
 
 /*
 ** ===================================================================
