diff -Nurd lua-5.3/src/ldo.c lua-5.3/src/ldo.c
--- lua-5.3/src/ldo.c	2020-09-30 08:27:32.770988404 +0300
+++ lua-5.3/src/ldo.c	2020-09-30 08:26:21.694346748 +0300
@@ -60,7 +60,7 @@
 	try { a } catch(...) { if ((c)->status == 0) (c)->status = -1; }
 #define luai_jmpbuf		int  /* dummy variable */
 
-#elif defined(LUA_USE_POSIX)				/* }{ */
+#elif defined(LUA_USE_ULONGJMP)				/* }{ */
 
 /* in POSIX, try _longjmp/_setjmp (more efficient) */
 #define LUAI_THROW(L,c)		_longjmp((c)->b, 1)
diff -Nurd lua-5.3/src/liolib.c lua-5.3/src/liolib.c
--- lua-5.3/src/liolib.c	2020-09-30 08:27:32.770988404 +0300
+++ lua-5.3/src/liolib.c	2020-09-30 08:26:21.694346748 +0300
@@ -54,7 +54,7 @@
 
 #if !defined(l_popen)		/* { */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_POPEN)      /* { */
 
 #define l_popen(L,c,m)		(fflush(NULL), popen(c,m))
 #define l_pclose(L,file)	(pclose(file))
@@ -103,7 +103,7 @@
 
 #if !defined(l_fseek)		/* { */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_FSEEKO)	/* { */
 
 #include <sys/types.h>
 
diff -Nurd lua-5.3/src/loslib.c lua-5.3/src/loslib.c
--- lua-5.3/src/loslib.c	2020-09-30 08:27:32.770988404 +0300
+++ lua-5.3/src/loslib.c	2020-09-30 08:26:21.694346748 +0300
@@ -81,7 +81,7 @@
 ** where it uses gmtime_r/localtime_r
 */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_GMTIME_R)   /* { */
 
 #define l_gmtime(t,r)		gmtime_r(t,r)
 #define l_localtime(t,r)	localtime_r(t,r)
@@ -108,7 +108,7 @@
 */
 #if !defined(lua_tmpnam)	/* { */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_MKSTEMP)    /* { */
 
 #include <unistd.h>
 
diff -Nurd lua-5.3/src/lstring.c lua-5.3/src/lstring.c
--- lua-5.3/src/lstring.c	2020-09-30 08:27:07.934764097 +0300
+++ lua-5.3/src/lstring.c	2020-09-30 08:26:21.694346748 +0300
@@ -134,12 +134,14 @@
   TString *ts;
   GCObject *o;
   size_t totalsize;  /* total size of TString object */
+  char *str;
   totalsize = sizelstring(l);
   o = luaC_newobj(L, tag, totalsize);
   ts = gco2ts(o);
   ts->hash = h;
   ts->extra = 0;
-  getstr(ts)[l] = '\0';  /* ending 0 */
+  str = getstr(ts);
+  str[l] = '\0';  /* ending 0 */
   return ts;
 }
 
diff -Nurd lua-5.3/src/luaconf.h lua-5.3/src/luaconf.h
--- lua-5.3/src/luaconf.h	2020-09-30 08:27:32.770988404 +0300
+++ lua-5.3/src/luaconf.h	2020-09-30 08:26:21.694346748 +0300
@@ -11,6 +11,7 @@
 #include <limits.h>
 #include <stddef.h>
 
+#include "localluaconf.h"
 
 /*
 ** ===================================================================
@@ -61,14 +62,12 @@
 #if defined(LUA_USE_LINUX)
 #define LUA_USE_POSIX
 #define LUA_USE_DLOPEN		/* needs an extra library: -ldl */
-#define LUA_USE_READLINE	/* needs some extra libraries */
 #endif
 
 
 #if defined(LUA_USE_MACOSX)
 #define LUA_USE_POSIX
 #define LUA_USE_DLOPEN		/* MacOS does not need -ldl */
-#define LUA_USE_READLINE	/* needs an extra library: -lreadline */
 #endif
 
 
