diff -Nurd lua-5.3/src/ldo.c lua-5.3/src/ldo.c
--- lua-5.3/src/ldo.c	2019-06-03 01:45:00.598378079 +0300
+++ lua-5.3/src/ldo.c	2018-11-09 23:58:35.688634650 +0200
@@ -60,7 +60,7 @@
 	try { a } catch(...) { if ((c)->status == 0) (c)->status = -1; }
 #define luai_jmpbuf		int  /* dummy variable */
 
-#elif defined(LUA_USE_POSIX)				/* }{ */
+#elif defined(LUA_USE_ULONGJMP)				/* }{ */
 
 /* in POSIX, try _longjmp/_setjmp (more efficient) */
 #define LUAI_THROW(L,c)		_longjmp((c)->b, 1)
diff -Nurd lua-5.3/src/liolib.c lua-5.3/src/liolib.c
--- lua-5.3/src/liolib.c	2019-06-03 01:45:00.598378079 +0300
+++ lua-5.3/src/liolib.c	2018-11-09 23:58:35.692634650 +0200
@@ -54,7 +54,7 @@
 
 #if !defined(l_popen)		/* { */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_POPEN)      /* { */
 
 #define l_popen(L,c,m)		(fflush(NULL), popen(c,m))
 #define l_pclose(L,file)	(pclose(file))
@@ -103,7 +103,7 @@
 
 #if !defined(l_fseek)		/* { */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_FSEEKO)	/* { */
 
 #include <sys/types.h>
 
diff -Nurd lua-5.3/src/loslib.c lua-5.3/src/loslib.c
--- lua-5.3/src/loslib.c	2019-06-03 01:45:00.598378079 +0300
+++ lua-5.3/src/loslib.c	2018-11-09 23:58:35.692634650 +0200
@@ -81,7 +81,7 @@
 ** where it uses gmtime_r/localtime_r
 */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_GMTIME_R)   /* { */
 
 #define l_gmtime(t,r)		gmtime_r(t,r)
 #define l_localtime(t,r)	localtime_r(t,r)
@@ -108,7 +108,7 @@
 */
 #if !defined(lua_tmpnam)	/* { */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_MKSTEMP)    /* { */
 
 #include <unistd.h>
 
diff -Nurd lua-5.3/src/luaconf.h lua-5.3/src/luaconf.h
--- lua-5.3/src/luaconf.h	2019-06-03 01:45:00.598378079 +0300
+++ lua-5.3/src/luaconf.h	2018-11-09 23:58:35.692634650 +0200
@@ -11,6 +11,7 @@
 #include <limits.h>
 #include <stddef.h>
 
+#include "localluaconf.h"
 
 /*
 ** ===================================================================
@@ -61,14 +62,12 @@
 #if defined(LUA_USE_LINUX)
 #define LUA_USE_POSIX
 #define LUA_USE_DLOPEN		/* needs an extra library: -ldl */
-#define LUA_USE_READLINE	/* needs some extra libraries */
 #endif
 
 
 #if defined(LUA_USE_MACOSX)
 #define LUA_USE_POSIX
 #define LUA_USE_DLOPEN		/* MacOS does not need -ldl */
-#define LUA_USE_READLINE	/* needs an extra library: -lreadline */
 #endif
 
 
