diff -Nurd lua-5.3/src/ldo.c lua-5.3/src/ldo.c
--- lua-5.3/src/ldo.c	2015-01-30 22:34:26.329033992 +0200
+++ lua-5.3/src/ldo.c	2015-02-14 22:51:25.740710930 +0200
@@ -60,7 +60,7 @@
 	try { a } catch(...) { if ((c)->status == 0) (c)->status = -1; }
 #define luai_jmpbuf		int  /* dummy variable */
 
-#elif defined(LUA_USE_POSIX)				/* }{ */
+#elif defined(LUA_USE_ULONGJMP)				/* }{ */
 
 /* in POSIX, try _longjmp/_setjmp (more efficient) */
 #define LUAI_THROW(L,c)		_longjmp((c)->b, 1)
diff -Nurd lua-5.3/src/liolib.c lua-5.3/src/liolib.c
--- lua-5.3/src/liolib.c	2015-01-30 22:34:26.329033992 +0200
+++ lua-5.3/src/liolib.c	2015-02-14 22:33:35.100734420 +0200
@@ -47,7 +47,7 @@
 
 #if !defined(l_popen)		/* { */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_POPEN)      /* { */
 
 #define l_popen(L,c,m)		(fflush(NULL), popen(c,m))
 #define l_pclose(L,file)	(pclose(file))
diff -Nurd lua-5.3/src/loslib.c lua-5.3/src/loslib.c
--- lua-5.3/src/loslib.c	2015-02-14 22:33:05.196735076 +0200
+++ lua-5.3/src/loslib.c	2015-02-14 22:32:40.800735611 +0200
@@ -58,7 +58,7 @@
 ** uses mkstemp.
 */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_MKSTEMP)	/* { */
 
 #include <unistd.h>
 
diff -Nurd lua-5.3/src/luaconf.h lua-5.3/src/luaconf.h
--- lua-5.3/src/luaconf.h	2015-02-14 22:33:05.196735076 +0200
+++ lua-5.3/src/luaconf.h	2015-02-14 22:32:40.800735611 +0200
@@ -11,6 +11,7 @@
 #include <limits.h>
 #include <stddef.h>
 
+#include "localluaconf.h"
 
 /*
 ** ===================================================================
@@ -61,14 +62,12 @@
 #if defined(LUA_USE_LINUX)
 #define LUA_USE_POSIX
 #define LUA_USE_DLOPEN		/* needs an extra library: -ldl */
-#define LUA_USE_READLINE	/* needs some extra libraries */
 #endif
 
 
 #if defined(LUA_USE_MACOSX)
 #define LUA_USE_POSIX
 #define LUA_USE_DLOPEN		/* MacOS does not need -ldl */
-#define LUA_USE_READLINE	/* needs an extra library: -lreadline */
 #endif
 
 
