diff -Nurd lua-5.4/src/ldo.c lua-5.4/src/ldo.c
--- lua-5.4/src/ldo.c	2023-06-08 05:45:43.164933363 +0300
+++ lua-5.4/src/ldo.c	2023-06-08 05:45:54.989049862 +0300
@@ -60,7 +60,7 @@
 	try { a } catch(...) { if ((c)->status == 0) (c)->status = -1; }
 #define luai_jmpbuf		int  /* dummy variable */
 
-#elif defined(LUA_USE_POSIX)				/* }{ */
+#elif defined(LUA_USE_ULONGJMP)				/* }{ */
 
 /* in POSIX, try _longjmp/_setjmp (more efficient) */
 #define LUAI_THROW(L,c)		_longjmp((c)->b, 1)
@@ -129,9 +129,9 @@
         lua_unlock(L);
         g->panic(L);  /* call panic function (last chance to jump out) */
       }
-      abort();
     }
   }
+  abort();
 }
 
 
diff -Nurd lua-5.4/src/liolib.c lua-5.4/src/liolib.c
--- lua-5.4/src/liolib.c	2023-06-08 05:45:43.164933363 +0300
+++ lua-5.4/src/liolib.c	2023-06-08 05:45:54.989049862 +0300
@@ -54,7 +54,7 @@
 
 #if !defined(l_popen)		/* { */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_POPEN)      /* { */
 
 #define l_popen(L,c,m)		(fflush(NULL), popen(c,m))
 #define l_pclose(L,file)	(pclose(file))
@@ -115,7 +115,7 @@
 
 #if !defined(l_fseek)		/* { */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_FSEEKO)	/* { */
 
 #include <sys/types.h>
 
diff -Nurd lua-5.4/src/loslib.c lua-5.4/src/loslib.c
--- lua-5.4/src/loslib.c	2023-06-08 05:45:43.164933363 +0300
+++ lua-5.4/src/loslib.c	2023-06-08 05:45:54.989049862 +0300
@@ -74,7 +74,7 @@
 ** where it uses gmtime_r/localtime_r
 */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_GMTIME_R)   /* { */
 
 #define l_gmtime(t,r)		gmtime_r(t,r)
 #define l_localtime(t,r)	localtime_r(t,r)
@@ -101,7 +101,7 @@
 */
 #if !defined(lua_tmpnam)	/* { */
 
-#if defined(LUA_USE_POSIX)	/* { */
+#if defined(LUA_USE_MKSTEMP)    /* { */
 
 #include <unistd.h>
 
diff -Nurd lua-5.4/src/luaconf.h lua-5.4/src/luaconf.h
--- lua-5.4/src/luaconf.h	2023-06-08 05:45:43.164933363 +0300
+++ lua-5.4/src/luaconf.h	2023-06-08 05:45:54.989049862 +0300
@@ -11,6 +11,7 @@
 #include <limits.h>
 #include <stddef.h>
 
+#include "localluaconf.h"
 
 /*
 ** ===================================================================
