diff -Nurd tolua-5.2/src/bin/lua/variable.lua tolua-5.2/src/bin/lua/variable.lua
--- tolua-5.2/src/bin/lua/variable.lua	2022-06-19 14:00:07.051434219 +0300
+++ tolua-5.2/src/bin/lua/variable.lua	2022-06-19 14:01:07.411765219 +0300
@@ -116,7 +116,10 @@
  -- check self value
  if class and static==nil then
 	 output('#ifndef TOLUA_RELEASE\n')
-  output('  if (!self) tolua_error(tolua_S,"invalid \'self\' in accessing variable \''..self.name..'\'",NULL);');
+	 output('  if (!self) {\n')
+	 output('    tolua_error(tolua_S,"invalid \'self\' in accessing variable \''..self.name..'\'",NULL);\n')
+	 output('    return 0;\n')
+	 output('  }\n')
 		output('#endif\n')
  end
 
@@ -163,7 +166,10 @@
 		output('#ifndef TOLUA_RELEASE\n')
 		output('  tolua_Error tolua_err;')
   if class and static==nil then
-   output('  if (!self) tolua_error(tolua_S,"invalid \'self\' in accessing variable \''..self.name..'\'",NULL);');
+   output('  if (!self) {\n')
+   output('    tolua_error(tolua_S,"invalid \'self\' in accessing variable \''..self.name..'\'",NULL);\n')
+   output('    return 0;\n')
+   output('  }\n')
   elseif static then
    _,_,self.mod = strfind(self.mod,'^%s*static%s%s*(.*)')
   end
diff -Nurd tolua-5.2/src/bin/tolua.c tolua-5.2/src/bin/tolua.c
--- tolua-5.2/src/bin/tolua.c	2022-06-19 14:00:07.051434219 +0300
+++ tolua-5.2/src/bin/tolua.c	2020-02-22 14:33:47.775736302 +0200
@@ -22,6 +22,9 @@
 #include <stdlib.h>
 #include <string.h>
 
+#ifndef LUA_SOURCE
+int tolua_tolua_open(lua_State *L);
+#endif
 
 static void help (void)
 {
@@ -106,7 +109,6 @@
 
 #ifndef LUA_SOURCE
  {
-  int tolua_tolua_open (lua_State* L);
   tolua_tolua_open(L);
  }
 #else
diff -Nurd tolua-5.2/src/lib/tolua_event.c tolua-5.2/src/lib/tolua_event.c
--- tolua-5.2/src/lib/tolua_event.c	2022-06-19 14:00:07.051434219 +0300
+++ tolua-5.2/src/lib/tolua_event.c	2022-03-19 10:33:00.694585520 +0200
@@ -18,6 +18,8 @@
 
 #include "tolua.h"
 
+#include "tolua_event.h"
+
 /* Store at peer
 	* It stores, creating the corresponding table if needed,
 	* the pair key/value in the corresponding peer table
@@ -391,8 +393,8 @@
       }
       else if (lua_isuserdata(L,-1) && *((void**)lua_touserdata(L,-1))==NULL) {
         /* free object */
-        free(u);
         tolua_release(L,u);                /* unmap from tolua tables */
+        free(u);
       }
     }
     lua_settop(L,top);
diff -Nurd tolua-5.2/src/lib/tolua_map.c tolua-5.2/src/lib/tolua_map.c
--- tolua-5.2/src/lib/tolua_map.c	2022-06-19 14:00:07.051434219 +0300
+++ tolua-5.2/src/lib/tolua_map.c	2020-02-22 14:33:47.779735894 +0200
@@ -21,6 +21,7 @@
 #include <stdlib.h>
 #include <math.h>
 
+int tolua_bnd_takeownership(lua_State *L);
 
 static char toluaname[128] = "tolua.";
 static const char* TOLUANAME (const char* n)  
